modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/hash/PageMemoryHashIndexStorage.java

```diff
@@ -329,4 +340,34 @@ public void startCleanup() {
     public void finishCleanup() {
         state.compareAndSet(StorageState.CLEANUP, StorageState.RUNNABLE);
     }
+
+    @Override
+    public @Nullable RowId getLastBuildRowId() {
+        return busy(() -> {
+            throwExceptionIfStorageInProgressOfRebalance(state.get(), this::createStorageInfo);
+
+            try {
+                UUID lastBuildRowIdUuid = indexMetaTree.findOne(new IndexMetaKey(indexDescriptor().id())).lastBuildRowIdUuid();
```

Comment:
I don't like that we always read it from the tree, access to cached version would be so much faster. The only time we really must read it is the start

Reply:
I do not think that the operation of building an index will often occur, and does it make sense to keep an empty value in memory? If you insist, I can do caching.

Reply:
I insist

Reply:
Try to fix it.